### PR TITLE
DAOS-16027 test: Adding daos_test REBUILD31-34 subtests

### PR DESCRIPTION
**Pull request by phender**
Adding rebuild tests currently not executed by doas_test/rebuild.py.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test-medium-md-on-ssd: false
Test-tag: test_rebuild_31 test_rebuild_32 test_rebuild_33 test_rebuild_34

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — COMMENTED** [review]
Do we really want to add another 4 * 800s timeout (Almost an hour) to `pr`?

**Review by daltonbohning — COMMENTED** [review]
FYI Test-tag wrapped
```
Test-tag: test_rebuild_31 test_rebuild_32 test_rebuild_33
test_rebuild_34
```

File: src/tests/ftest/daos_test/rebuild.yaml [hunk 1]
```diff
@@ -66,6 +72,10 @@ daos_tests:
     test_rebuild_28: DAOS_Rebuild_28
     test_rebuild_29: DAOS_Rebuild_29
     test_rebuild_30: DAOS_Rebuild_30
+    test_rebuild_31: DAOS_Rebuild_31
+    test_rebuild_32: DAOS_Rebuild_32
+    test_rebuild_33: DAOS_Rebuild_33
+    test_rebuild_34: DAOS_Rebuild_34
```
Comment (kccain) [inline src/tests/ftest/daos_test/rebuild.yaml:78]: I think we'll need a test_rebuild_35 too.

File: src/tests/ftest/daos_test/rebuild.yaml [hunk 2]
```diff
@@ -66,6 +72,10 @@ daos_tests:
     test_rebuild_28: DAOS_Rebuild_28
     test_rebuild_29: DAOS_Rebuild_29
     test_rebuild_30: DAOS_Rebuild_30
+    test_rebuild_31: DAOS_Rebuild_31
```
Comment (kccain) [inline src/tests/ftest/daos_test/rebuild.yaml:75]: Unfortunately test_rebuild_31 currently hangs. I discussed with @liuxuezhao and that failure is related to some overall rebuild error case handling that needs refinement, https://daosio.atlassian.net/browse/DAOS-15847.

He gave me an idea to try with test_rebuild_31 using a different type of fault injection. I can check that out separately. In the meantime, maybe we could (if the tests are able to pass) enable tests 32-35.

**Comment by phender** [comment]
Skip list entries added for the new tests:
```
[['DAOS-16575', 'test_method_name', 'test_rebuild_31']]
[['DAOS-16576', 'test_method_name', 'test_rebuild_32']]
```

**Review by daltonbohning — COMMENTED** [review]
Can we put these in daily instead of pr? this adds 4 tests * 800s each. Which is almost 1 hour for every pr

**Comment by phender** [comment]
> Can we put these in daily instead of pr? this adds 4 tests * 800s each. Which is almost 1 hour for every pr

@kccain thoughts?

**Comment by phender** [comment]
Rebuild 31-32 skipped and Rebuild 33-34 passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14584/4/testReport/FTEST_daos_test/DaosCoreTestRebuild/

File: src/tests/ftest/daos_test/rebuild.yaml (hunk 2 above)
Comment (phender) [reply src/tests/ftest/daos_test/rebuild.yaml:79]: I've added skip list entries for test_rebuild_31 and test_rebuild_32.

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
Tests passing/skipping as expected in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14584/5/testReport/FTEST_daos_test/DaosCoreTestRebuild/

**Pull request merged**